src/libstrongswan/plugins/botan/botan_util_keys.c

```diff
@@ -232,6 +265,18 @@ private_key_t *botan_private_key_load(key_type_t type, va_list args)
 	{
 		this = botan_ed_private_key_adopt(key);
 	}
+	else
+#endif
+#ifdef BOTAN_HAS_ML_DSA
+	if (streq(name, "ML-DSA") && (type == KEY_ANY || type == KEY_ML_DSA_44 ||
+							  type == KEY_ML_DSA_65 || type == KEY_ML_DSA_87))
+	{
+		if (type == KEY_ANY)
+		{
+			type = key_type_from_pkcs8(blob);
```

Comment:
What if the type can't be determined? `botan_ml_dsa_private_key_adopt()` should probably fail if `type` is `KEY_ANY`.

Reply:
botan parses the public key already in line 255 and exits if the key type is not supported:
```
	if (botan_privkey_load(&key, rng, blob.ptr, blob.len, NULL))
	{
		botan_rng_destroy(rng);
		return NULL;
	}
```
Our problem is that the botan **`key`** object currently doesn't offer a method to extract which type of ML-DSA key was loaded (according to the source code this is on the TODO list for botan 4). Therefore we have to do our own PKCS#8 parsing to determine the key type. Since the **`ml`** and **`wolfssl`** plugins are currently not able to do their own PKCS#8 parsing of an ASN.1 encoded ML-DSA private key, a **`private_key_info_decode()`** function would be practical with the long term goal of getting rid of the **`pkcs1`** and **`pkcs8`** plugins, altoghether.

Reply:
Similar to the discussion above with the public key, would it make sense to compare or adopt the actually parsed type (if it's not `KEY_ANY`, just to make this future-proof if there will ever be other OIDs)?